include/functions.h

```diff
@@ -959,157 +959,45 @@ s32 DynaPolyActor_IsInRidingRotatingState(DynaPolyActor* dynaActor);
-s32 Camera_ModeFIXD1(Camera* camera);
-s32 Camera_ModeFIXD2(Camera* camera);
-s32 Camera_ModeFIXD3(Camera* camera);
-s32 Camera_ModeFIXD4(Camera* camera);
-s32 Camera_ModeFIXD0(Camera* camera);
-s32 Camera_ModeSUBJ1(Camera* camera);
-s32 Camera_ModeSUBJ2(Camera* camera);
-s32 Camera_ModeSUBJ3(Camera* camera);
-s32 Camera_ModeSUBJ4(Camera* camera);
-s32 Camera_ModeSUBJ0(Camera* camera);
-s32 Camera_ModeDATA0(Camera* camera);
-s32 Camera_ModeDATA1(Camera* camera);
-s32 Camera_ModeDATA2(Camera* camera);
-s32 Camera_ModeDATA3(Camera* camera);
-s32 Camera_ModeDATA4(Camera* camera);
-s32 Camera_ModeUNIQ1(Camera* camera);
-s32 Camera_ModeUNIQ2(Camera* camera);
-s32 Camera_ModeUNIQ3(Camera* camera);
-s32 Camera_ModeUNIQ4(Camera* camera);
-s32 Camera_ModeUNIQ5(Camera* camera);
-s32 Camera_ModeUNIQ0(Camera* camera);
-s32 Camera_ModeUNIQ6(Camera* camera);
-s32 Camera_ModeUNIQ7(Camera* camera);
-s32 Camera_ModeUNIQ8(Camera* camera);
-s32 Camera_ModeUNIQ9(Camera* camera);
-s32 Camera_ModeDEMO1(Camera* camera);
-s32 Camera_ModeDEMO2(Camera* camera);
-s32 Camera_ModeDEMO3(Camera* camera);
-s32 Camera_ModeDEMO4(Camera* camera);
-s32 Camera_ModeDEMO5(Camera* camera);
-s32 Camera_ModeDEMO6(Camera* camera);
-s32 Camera_ModeDEMO7(Camera* camera);
-s32 Camera_ModeDEMO8(Camera* camera);
-s32 Camera_ModeDEMO9(Camera* camera);
-s32 Camera_ModeDEMO0(Camera* camera);
-s32 Camera_ModeSPEC0(Camera* camera);
-s32 Camera_ModeSPEC1(Camera* camera);
-s32 Camera_ModeSPEC2(Camera* camera);
-s32 Camera_ModeSPEC3(Camera* camera);
-s32 Camera_ModeSPEC4(Camera* camera);
-s32 Camera_ModeSPEC5(Camera* camera);
-s32 Camera_ModeSPEC6(Camera* camera);
-s32 Camera_ModeSPEC7(Camera* camera);
-s32 Camera_ModeSPEC8(Camera* camera);
-s32 Camera_ModeSPEC9(Camera* camera);
-Camera* Camera_Alloc(View* view, CollisionContext* bg, PlayState* play);
-void Camera_Free(Camera* camera);
-void Camera_Init(Camera* camera, View* view, CollisionContext* bg, PlayState* play);
-// void func_800DDFE0(void);
+Camera* Camera_Create(View* view, CollisionContext* colCtx, PlayState* play);
+void Camera_Destroy(Camera* camera);
+void Camera_Init(Camera* camera, View* view, CollisionContext* colCtx, PlayState* play);
+void func_800DDFE0(Camera* camera);
 void Camera_InitPlayerSettings(Camera* camera, Player* player);
 s32 Camera_ChangeStatus(Camera* camera, s16 status);
-// void func_800DE324(void);
-// void func_800DE62C(void);
-// void func_800DE840(void);
-// void func_800DE890(void);
-// UNK_TYPE4 func_800DE954(Camera* camera);
-Vec3s* Camera_Update(Vec3s* param_1, Camera* camera);
-// void func_800DF498(void);
-u32 Camera_SetMode(Camera* camera, s16 mode, s8 param_3);
+s32 Camera_UpdateWater(Camera* camera);
+void Camera_EarthquakeDay3(Camera* camera);
+s32 Camera_UpdateHotRoom(Camera* camera);
+s32 Camera_SetSwordDistortion(Camera* camera);
+s32 Camera_RequestGiantsMaskSetting(Camera* camera);
+Vec3s* Camera_Update(Vec3s* inputDir, Camera* camera);
+s32 Camera_SetModeSettingStateFlags(Camera* camera);
```

Comment:
This is very wordy. Not a fan of the double use of set.

Waht about `SetModeUpdatingStateFlags`

Reply:
Probably a better name would be related to how it sets `CAM_STATE_2` and unsets `CAM_STATE_12`, so it might just be best to leave it unnamed until those two stateflags are understood

Reply:
Just to make a note for later, this function specifically does:
```c
    Camera_SetStateFlag(camera, CAM_STATE_3 | CAM_STATE_2); // CAM_STATE_3 is set only immediately to be unset
    Camera_UnsetStateFlag(camera, CAM_STATE_12 | CAM_STATE_3);
```
So `CAM_STATE_3` is set and unset, so I guess you can say it's always unset.
`CAM_STATE_12` is completely unused except for being unset here, so that's not relevant.
`CAM_STATE_2` is for resetting `nextBgCamIndex`, but there's a lot of state flags dealing with bgCam Indices, so there still needs more work figuring out the subtle differences for those bgCam-related stateflags. But it has something to do with that.